src/Microsoft.TestPlatform.Common/ExtensionFramework/Utilities/TestExtensions.cs

```diff
@@ -385,21 +385,21 @@ internal TestExtensions GetExtensionsDiscoveredFromAssembly(string extensionAsse
     /// Gets the cached extensions for the current process.
     /// </summary>
     /// 
-    /// <returns>A map representing the cached extensions for the current process.</returns>
+    /// <returns>A dictionary representing the cached extensions for the current process.</returns>
     internal Dictionary<string, HashSet<string>> GetCachedExtensions()
     {
-        var extensionMap = new Dictionary<string, HashSet<string>>();
+        var extensions = new Dictionary<string, HashSet<string>>();
 
         // Write all "known" cached extension.
-        AddCachedExtensionToExtensionMap(extensionMap, "TestDiscoverers", TestDiscoverers?.Values.ToList());
-        AddCachedExtensionToExtensionMap(extensionMap, "TestExecutors", TestExecutors?.Values.ToList());
-        AddCachedExtensionToExtensionMap(extensionMap, "TestExecutors2", TestExecutors2?.Values.ToList());
-        AddCachedExtensionToExtensionMap(extensionMap, "TestSettingsProviders", TestSettingsProviders?.Values.ToList());
-        AddCachedExtensionToExtensionMap(extensionMap, "TestLoggers", TestLoggers?.Values.ToList());
-        AddCachedExtensionToExtensionMap(extensionMap, "TestHosts", TestHosts?.Values.ToList());
-        AddCachedExtensionToExtensionMap(extensionMap, "DataCollectors", DataCollectors?.Values.ToList());
-
-        return extensionMap;
+        AddCachedExtensionToDictionary(extensions, "TestDiscoverers", TestDiscoverers?.Values.ToList());
```

Comment:
Sorry I should have called it out in the previous review (feel free to ignore it, the perf impact is most likely neglectable here). The `ToList` call is no longer necessary as you now allow `IEnumerable`.

Reply:
Oh, that makes sense, many thanks for pointing it out. I'll need to push another commit either way since the current build has finished in the meantime but it didn't report the status back to Github so it appears as if it's still running ...